src/Sulu/Bundle/CategoryBundle/Entity/CategoryRepository.php

```diff
@@ -11,151 +11,97 @@
-            }
-            if ($depth !== null) {
-                $queryBuilder->andWhere('category.depth = :depth');
-            }
-
-            if ($sortBy) {
-                $sortOrder = ($sortOrder) ? $sortOrder : 'asc';
-                $queryBuilder->addOrderBy('category.' . $sortBy, $sortOrder);
-            }
-
-            $query = $queryBuilder->getQuery();
-            if ($parent !== null) {
-                $query->setParameter('parentId', $parent);
-            }
-            if ($depth !== null) {
-                $query->setParameter('depth', $depth);
-            }
-
-            return $query->getResult();
-        } catch (NoResultException $ex) {
-            return;
-        }
+        $queryBuilder->where($queryBuilder->expr()->in('category.id', ':ids'));
+        $queryBuilder->setParameter('ids', $ids);
+
+        return $queryBuilder->getQuery()->getResult();
     }
 
     /**
-     * Returns the children for a given category.
-     *
-     * @param int         $key       the key of the category to return the children for
-     * @param string|null $sortBy    column name to sort by
-     * @param string|null $sortOrder sort order
-     *
-     * @return Category[]
+     * {@inheritdoc}
      */
-    public function findChildren($key, $sortBy = null, $sortOrder = null)
+    public function findCategoriesByParentId($parentId = null)
     {
-        try {
-            $queryBuilder = $this->getCategoryQuery()
-                ->from('SuluCategoryBundle:Category', 'parent')
-                ->andWhere('parent.key = :key')
-                ->andWhere('category.parent = parent');
-
-            if ($sortBy) {
-                $sortOrder = ($sortOrder) ? $sortOrder : 'asc';
-                $queryBuilder->addOrderBy('category.' . $sortBy, $sortOrder);
-            }
-
-            $query = $queryBuilder->getQuery();
-            $query->setParameter('key', $key);
-
-            return $query->getResult();
-        } catch (NoResultException $ex) {
-            return;
+        $queryBuilder = $this->getCategoryQuery();
+
+        if ($parentId === null) {
+            $queryBuilder->andWhere('category.parent IS NULL');
+        } else {
+            $queryBuilder->andWhere('categoryParent.id = :parentId');
+        }
+
+        $query = $queryBuilder->getQuery();
+
+        if ($parentId !== null) {
+            $query->setParameter('parentId', $parentId);
         }
+
+        return $query->getResult();
     }
 
     /**
-     * Finds the categories with the given ids.
-     *
-     * @param array $ids The ids to load
-     *
-     * @return Category[]
+     * {@inheritdoc}
      */
-    public function findCategoryByIds(array $ids)
+    public function findCategoriesBetween($fromIds, $toIds)
     {
-        try {
-            if (empty($ids)) {
-                throw new NoResultException();
-            }
+        $queryBuilder = $this->getCategoryQuery();
```

Comment:
This is the repository, not the manager. And yes, I would say that we should do it this way, because in this case we do not need most of the joins being done here.